### PR TITLE
Added hoarder variant to six houses

### PR DESCRIPTION
**Pull request by tjole**
#### Summary
Content "Added hoarder variants to houses"

#### Purpose of change
No related issue. Adds more mapgen content.

The homes of hoarders are often cluttered and hard to move through. In-game, this is represented by impassable/sight-blocking furniture and a large amount of item spawns. The furniture limits line-of-sight and provides cover for enemies to move around, forcing combat to be much closer.

#### Describe the solution
Added hoarder itemgroup and mapgen palette to support adding this variant to multiple houses. Each variant is weighted to 1% of the base variant.

#### Describe alternatives you've considered
Considered not adding this, or not using the term 'hoarder', as hoarding disorder is a mental health issue and a negative portrayal of this would be unfair to people with the condition.

#### Testing
Spawned each house variant from the debug menu and confirmed that item/monster spawns are as expected and terrain under furniture is correct type.

house_23
house_40

### CODE REVIEW

**Comment by LurchingOtter** [comment]
Is there a way to add rotten food to the itemgroup? I'd like to add a small number of unsealed/rotten food items, if possible.

**Comment by PatrikLundell** [comment]
I'd suggest having a very low weight of this spawn. 5% seems excessively high.

**Comment by toweroglass** [comment]
> I'd suggest having a very low weight of this spawn. 5% seems excessively high.

one in twenty is possibly a little high, but I wouldn't make it crazy rare. this pretty common IRL to varying extents.

**Comment by curstwist** [comment]
I had begun a similar project to add in chances of abandoned houses if you'd like to see that approach.  It uses a palette distribution instead of a new map variant.  If you'd like to talk about this approach and implementing a large scale system using palette distribution let me know.  The biggest challenge is the need to standardize the symbols in all house maps and then link each into the palette sets (this is partially done from when I moved most to the standard palette for houses).  

One long term goal I have is moving every house map onto this palette system.
My sample PR  "Add abandoned house palette, add new variable parameters for houses #55570" may be useful to look at.

If you stick the variant approach instead (which if fine), I suggest leaning on your palette more by moving as much of the object entries (terrain, furniture, items, etc) into the palette as possible.  You can also call on multiple palettes in mapgen, so you can use the ones that can add linoleum to bathrooms `standard_domestic_lino_bathroom` and `standard_domestic_landscaping_palette`.  I should probably also add an overgrown version of the landscaping palette now that I think about it.

If you switch approaches, you'd have to adjust how you are spawning the extra furniture (probably by just adding the new arrays into a bunch of existing symbols and the floor symbols with a chance for `null` to leave some empty space (this would give the benefit of having a higher amount of variability in what spawns where and also the open path, or lack, through the house).

I wanted to let you know about this option since it is still pretty new, but you aren't obligated to use it. Sorry this is all so late, I've been taking a break from development.

**Comment by LurchingOtter** [comment]
Thanks for all the feedback & suggestions.

RE spawn rate, I'll lower this to 1%. I think the number of people who experience some level of hoarding disorder is up to 5%, depending on the study. That being said, that's 'experience some level' as opposed to 'experience a severe level'. With that in mind, I think a 1% rate would be reasonable. Happy to listen to further suggestions on that, though.

@toweroglass
The trash piles are a great idea! Looking at the existing ones, though, I could only find one that had a description for the outdoors ("topped with dirt and grass"). I'd need to add a new terrain type for indoor clutter. After curstwist's comment, I'm going to add your suggestion under a new PR, with the aim of implementing a 'cluttered house' palette (as well as adding it to the hoarder palette).

@curstwist
Thanks for the suggestions - it's helpful to see an example of this being done. The only issue I could think of with using a palette was that randomly placing furniture throughout the house wouldn't guarantee certain things like 'goat trails', or paths to various parts of the houses. I'm not sure, outside of a variant with specific placings, how that could be done.

What I'd like to do is go ahead and get this merged, then start work on splitting out everything possible into palettes. I was thinking that perhaps I could introduce a 'cluttered' palette that doesn't have quite as extreme a level of items/furniture, and then extend that for the specific cases of the hoarding houses, which would still require variants, I think.

File: data/json/itemgroups/collections_domestic.json [hunk 1]
```diff
@@ -1187,5 +1187,22 @@
...
+      { "group": "dishes_utility", "prob": 10 },
+      { "group": "child_items", "prob": 10 },
+      { "group": "SUS_bathroom_sink", "prob": 10 },
+      { "group": "bed", "prob": 10 },
+      { "group": "games", "prob": 5 },
+      { "group": "SUS_junk_drawer", "prob": 1 }
+    ]
   }
-]
+]
```
Comment (anothersimulacrum) [inline data/json/itemgroups/collections_domestic.json:1208]:
```suggestion
]

```

File: data/json/mapgen_palettes/house_variant_palette.json [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+      "E": { "item": "hoarder", "chance": 80, "repeat": [2, 3] },
+      "H": { "item": "hoarder", "chance": 80, "repeat": [2, 3] },
+      "!": { "item": "hoarder", "chance": 80, "repeat": [ 5, 8 ] },
+      "~": { "item": "hoarder", "chance": 80, "repeat": [ 4, 6 ] },
+      "?": { "item": "hoarder", "chance": 80, "repeat": [ 5, 10 ] },
+      "=": { "item": "hoarder", "chance": 80, "repeat": [ 4, 8 ] },
+      "_": { "item": "hoarder", "chance": 90, "repeat": [ 2, 4 ] }
+    }
+  }
+]
```
Comment (anothersimulacrum) [inline data/json/mapgen_palettes/house_variant_palette.json:312]:
```suggestion
]

```

**Review by anothersimulacrum — COMMENTED** [review]
Can we get pictures?

**Pull request merged**